core/services/src/main/java/org/exoplatform/services/attachments/service/AttachmentServiceImpl.java

```diff
@@ -235,46 +260,86 @@ public void deleteAttachmentItemById(long userIdentityId,
+      Permission attachmentACL = new Permission(canView, canDelete, canEdit);
+
+      attachment.setAcl(attachmentACL);
+    } catch (Exception e) {
+      throw new ObjectNotFoundException("Can't convert attachment JCR node with id " + attachmentId + " to entity");
+    } finally {
+      if (session != null) {
+        session.logout();
       }
     }
-    return attachments;
+    return attachment;
   }
 
   @Override
-  public Attachment getAttachmentById(String attachmentId, SessionProvider sessionProvider) throws Exception {
+  public void moveAttachmentToNewPath(long userIdentityId,
+                                      String attachmentId,
+                                      String newPathDrive,
+                                      String newPath,
+                                      String entityType,
+                                      long entityId) throws ObjectNotFoundException, IllegalAccessException {
+    if (userIdentityId <= 0) {
+      throw new IllegalAccessException("User identity must be positive");
+    }
+    if (StringUtils.isEmpty(attachmentId)) {
+      throw new IllegalArgumentException("Attachment id is mandatory");
+    }
+    if (StringUtils.isEmpty(newPathDrive)) {
+      throw new IllegalArgumentException("New destination path's drive must not be empty");
+    }
+    if (!canEdit(userIdentityId, entityType, entityId)) {
+      throw new IllegalAccessException("User with identity id" + userIdentityId + "is not allowed to move attachment with id"
+          + attachmentId);
+    }
+
     Session session = null;
     try {
-      String workspace = repositoryService.getCurrentRepository().getConfiguration().getDefaultWorkspaceName();
-      session = sessionProvider.getSession(workspace, repositoryService.getCurrentRepository());
-      return EntityBuilder.fromAttachmentNode(repositoryService,
-                                              documentService,
-                                              workspace,
-                                              session,
-                                              attachmentId);
+      session = Utils.getSession(sessionProviderService, repositoryService);
+      Node attachmentNode = session.getNodeByUUID(attachmentId);
+      Node destNode = Utils.getParentFolderNode(session,
+                                                manageDriveService,
+                                                nodeHierarchyCreator,
+                                                nodeFinder,
+                                                newPathDrive,
+                                                newPath);
+      String destPath = destNode.getPath().concat("/").concat(attachmentNode.getName());
+      session.move(attachmentNode.getPath(), destPath);
+      session.save();
+    } catch (Exception e) {
+      throw new ObjectNotFoundException("Error while trying to move attachment node with id " + attachmentId + " to the new path "
```

Comment:
not compatible exceptions, can't transform all `Exception` => `ObjectNotFoundException`